text/0000-external-control-plane-gluster.md

```diff
@@ -0,0 +1,407 @@
+the task state will remain as "QUEUED" and node state will be changed
+to "UNREACHABLE".
+
+![Task Lifecycle](images/state-life-cycle.jpg)
+
+### Communication between Control Server and Node agent:
+
+The control server will not have access to node agents; Node agents
+will connect to the Control server and gets the details. Node agents
+can establish a WebSocket connection so that the Control server can
+push the notification when needed.
+
+
+## Monitoring and Dashboard
+
+The dashboard of the Control plane shows the real-time status of the
+cluster. Visualizing historical data is possible by using Prometheus
+or other similar solutions.
+
+The control plane will have up to date details about the health of the
+cluster, Volume utilization, and other metrics. The control plane will
+not store the historical data, and Prometheus can be used for
+visualizing historical data.
+
+Control plane can become Prometheus exporter by exposing a new API
+`/metrics`. Prometheus server running in the same node as the control
+plane can pick up the metrics from the control plane. Use the Grafana
+dashboard to visualize the metrics from Prometheus. (Note: Grafana
+dashboards can be embedded into any website, may be embedded in the
+WebUI of control plane)
+
+## Backup and Recovery of Cluster metadata
+
+Every change to Cluster meta information will be versioned. On a
+handshake, each node agent downloads the cluster meta information and
+saves it locally. If the control server node goes down, then it can be
+started in some other node by restoring data from any node agent(Or
+choose the node agent which contains the latest cluster data).
+
+## High availability of Control plane
+
+Control Server is no different from a web server, use all the
+available scaling and load balancing techniques for hosting it. Use
+the "Streaming Replication" feature of Postgresql for disaster
+recovery.
+
+More on this topic TBD. High availability of the database is also
+possible by using the Gluster Volume as storage for the
+database. (This is based on the fact that Replica 3 Gluster volume can
+be run as service without the need of Glusterd)
+
+## Usage:
+
+CLI commands used in the following examples. CLI is one of the
+mechanisms to control the Cluster.  Other ways to communicate with the
+Control plane are Web UI, ReST APIs, etc.
+
+### User management
+
+In the case of the self-hosted Control server, the first user will
+become the Admin user. Create more users using admin privileges.
+
+Granular user permissions that enable them to access specific
+resources on the Cluster, such as Volumes, Nodes, and Storage
+devices.
+
+Log in to the Control server by running the following command, and the
+token will be saved in `~/.gluster/token.json` on a successful login.
+
+All consecutive commands will use the token saved in the file
+mentioned above. To logout, run the following command.
+
+```
+$ glustercli login
```

Comment:
you want to call this tool `glustercli` or something else? like `moana` (means sea/kadalu/ocean) or something similar?